packages/util/src/bytes.ts

```diff
@@ -225,6 +225,19 @@ export function bigIntToBuffer(num: bigint) {
   return toBuffer('0x' + num.toString(16))
 }
 
+/**
+ * Converts a {@link bigint} to a {@link Buffer} for devp2p rlp encodings
+ * excluding `0` as rlp has its own special encoding for the same
+ * see: https://github.com/ethereumjs/ethereumjs-monorepo/issues/2402
+ */
+export function devP2PBigIntToBuffer(input: bigint | number): Buffer {
+  const inputNum = BigInt(input)
+  if (inputNum === 0n) {
```

Comment:
Just a reminder here (not directly relevant any more): we have decided on a policy to never use the *n notation for now for BigInt but always use `BigInt(*)` for compatibility reasons.